File: src/objects/DataOfLetter.java
package objects;

public class DataOfLetter {
   private String letter;
   private boolean shouldRotate;
   private int indexOfNextLetter;

   public DataOfLetter(String letter, boolean shouldRotate) {
      this.letter = letter;
      this.shouldRotate = shouldRotate;
   }

   public boolean isShouldRotate() {
      return shouldRotate;
   }

   public int getIndexOfNextLetter() {
      return indexOfNextLetter;
   }

   public void setIndexOfNextLetter(int indexOfNextLetter) {
      this.indexOfNextLetter = indexOfNextLetter;
   }

   public String getLetter() {
      return letter.toUpperCase();
   }

   public void setLetter(String letter) {
      this.letter = letter;
   }

   public boolean ShouldRotate() {
      return shouldRotate;
   }

   public void setShouldRotate(boolean shouldRotate) {
      this.shouldRotate = shouldRotate;
   }
}

File: DataOfLetter.cpp
#include <cctype>
#include <iostream>
#include <cstring>
#include <vector>
#include <algorithm>
using namespace std;

class DataOfLetter {
private:
	string letter;
	bool shouldRotate;
	int indexOfNextLetter;

public:
	DataOfLetter(string letter, bool shouldRotate) {
		this->letter = letter;
		this->shouldRotate = shouldRotate;
	}

	int getIndexOfNextLetter() {
		return this->indexOfNextLetter;
	}
	void setIndexOfNextLetter(int indexOfNextLetter) {
		this->indexOfNextLetter = indexOfNextLetter;
	}
	string getLetter() {
		string upperCaseLetter = this->letter;
		transform(upperCaseLetter.begin(), upperCaseLetter.end(), upperCaseLetter.begin(), ::toupper);
		return upperCaseLetter;
	}
	bool ShouldRotate() {
		return this->shouldRotate;
	}
};